src/SomeClass.java

```diff
@@ -0,0 +1,44 @@
+import java.util.Arrays;
+import java.util.List;
+
+public class SomeClass {
+
+    public static <T extends Number> void evenNumberCount(List<T> numbers){
+        int evenNumbers = 0;
+        for (T num : numbers) {
+            if (num instanceof Integer && num.intValue() % 2 == 0){

```

Comment:
Almost. What if I pass list of doubles to the method?
`SomeClass.evenNumberCount(Arrays.asList(2.0, 3.0, 4.0))`
I'd expect to see `2` as a result.

Reply:
I found my mistake and switched code to " if (num.doubleValue() % 2 == 0) ", according to my few tests, now it works correctly. 
Should I make a new commit with changes?

Reply:
Yep, why not? :)